### PR TITLE
[BUG] Fixing wrong output activation in MLPClassifier

### LINKED ISSUE #1596: [BUG] Wrong output activation in MLPClassifier
### Describe the bug

The activation used in MLPClassifier at the last layer is using the self.activation, however this parameter is for internal activation, the output activation should be softmax

### Steps/Code to reproduce the bug

_No response_

### Expected results

output probas sum to 1

### Actual results

output 0/1

### Versions

_No response_

### PR DESCRIPTION
**Pull request by hadifawaz1999**
Fix #1596

### CODE REVIEW

**Review by TonyBagnall — APPROVED** [review]
LGTM

File: aeon/classification/deep_learning/_mlp.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def build_model(self, input_shape, n_classes, **kwargs):
         input_layer, output_layer = self._network.build_network(input_shape, **kwargs)
 
         output_layer = keras.layers.Dense(
-            units=n_classes, activation=self.activation, use_bias=self.use_bias
```
Comment (lngdet) [inline aeon/classification/deep_learning/_mlp.py:169]: This same issue is present in `CNNClassifier`, see [here](https://github.com/aeon-toolkit/aeon/blob/708db2e76914d78c1d36044d73b47ad23941996a/aeon/classification/deep_learning/_cnn.py#L205-L207).

Comment (hadifawaz1999) [reply aeon/classification/deep_learning/_mlp.py:169]: This was actually intentional in the case of CNNs because the original authors proposed to use sigmoid as activation at the last layer, and simply do a mean squared error loss, instead of cross entropy (when the activation is softmax)

Comment (hadifawaz1999) [reply aeon/classification/deep_learning/_mlp.py:169]: as you see the default loss in case of CNNclassifier its not cross entropy its mse

Comment (lngdet) [reply aeon/classification/deep_learning/_mlp.py:169]: Thanks for the feedback, @hadifawaz1999
